### ISSUE TITLE
MSSQL Server. Authenticating with domain users

### ISSUE THREAD

**Issue by pbuentam** [grafana/grafana#19795]
**What would you like to be added**:
Being able to authenticate with AD users

**Why is this needed**:
In our Corporation all MSSQL users are AD users. When we try to connect we get the following error:
Login error: mssql: Login failed. The login is from an untrusted domain and cannot be used with Windows authentication.

It's the same error you get when using jDTS if not properly configured. With that jdbc driver you have to pass de domain, user, and usentlmv2 parameters with the conection url.

**Comment by marefr** [comment]
Duplicate of #11876

**Comment by pbuentam** [comment]
I think it is not a duplivate of #11876 as integrated security uses the identity of the process running Grafana. Wi have thousands of MSSQL instances so we need to provide the identity through the datasource. Looking into mssql driver documentation it is usually called "Active Directory authentication by password"

**Comment by marefr** [comment]
Sure. I've reopened the issue. However, I think this is currently blocked by https://github.com/denisenkom/go-mssqldb/issues/146 which is the library Grafana uses to connect to MSSQL.

**Comment by petemit** [comment]
It looks like they merged this in the past month. 

Is addressing this as simple as pulling in the new sql driver?

**Comment by marefr** [comment]
@petemit it could be. We're also using [Xorm](https://xorm.io/) and unless it doesn't do something special with connectionstring should be fine. We're currently using this version/commit of the library: https://github.com/denisenkom/go-mssqldb/commit/2be1aa521ff4499e74b7861a2779ba1e96e3e2c5 so also need to review the changes made since that up until latest master. Would suggest open a PR and take it from there.

**Comment by fifofonix** [comment]
This issue is present in our environment, and can be replicated when running a clone of denisenkom/go-mssqldb@2be1aa5, and can equally be shown to be fixed on denisenkom/go-mssqldb@master.  Our test:
```
% export SQLSERVER_DSN=sqlserver://hostname?user+id=<DOMAIN>\\<user>\&encrypt=true\&password=<url escaped password>\&TrustServerCertificate=true
% go test
```

**Comment by marefr** [comment]
@fifofonix thanks. Interested in contributing by opening a pull request to upgrade go-mssqldb library?

I see you're using `TrustServerCertificate`, maybe related to #24589.

**Comment by marefr** [comment]
Oops just notice you commented on #24589 :+1:

**Comment by dp106** [comment]
Can I ask is this still being investigated?  At the moment the only way I can connect to a SQL database is to create a SQL Login on every server I want to connect to. As well as being against our company policy this is impractical for connecting to multiple servers. I would like to be able to create an account in active directory that I can add to the datasource and then give that one account permissions to the relevant SQL servers we want to connect to.

**Comment by jothoma1** [comment]
Any news on this ?

**Comment by Quazmoz** [comment]
Any update on this? I would love to see it fixed

**Comment by oscarkilhed** [comment]
The MSSQL driver has been updated in https://github.com/grafana/grafana/pull/73597 is this still issue still present in 10.2?

**Comment by Quazmoz** [comment]
https://community.grafana.com/t/grafana-support-for-mssql-authentication/30643/2
It does not work for me (docker container running grafana-oss:10.2.2-ubuntu)
![image](https://github.com/grafana/grafana/assets/32716387/11ec43a3-e8fa-42ee-8386-0bf1388a164d)
![image](https://github.com/grafana/grafana/assets/32716387/2decd838-4fd6-4124-a991-42a5d547b270)

**Comment by fifofonix** [comment]
My understanding is that this issue has principally concerned the usage of domain users in SQLServer connection definitions. This did not work originally when this issue was opened but has been operational for us in production without issue since at least Grafana 9.x. We are running docker containers on linux (no kerberos needed anywhere) and choosing `SQL Server Authentication` in the screenshot above. Some may consider this a confusing name for the option but the descriptive text makes is clear that this supports `DOMAIN\User`, ie. AD domain users. No need for a SQLServer specific local account at all.

There is/was a separate/related issue with respect to getting proper verified SSL connectivity that related to the location go was checking when performing its certificate verification. I have posted on that separately (referenced above) and we still carry that minor mod in our docker containers whether or not it is really needed at this point.

**Comment by clasch1985** [comment]
I sumbled upon this issue this week. We have grafana enterprise on an OpenShift Cluster. I can connect with SQL Server Login, but I cannot connect with an AD user like DOMAIN\User and Password. I recieve following error:

`mssql: login error: Login failed. The login is from an untrusted domain and cannot be used with Integrated authentication.`

This sounds like a Kerberos issue. To be sure we have actived NTLM on SQL Server for testing purposes as we are not allowed to use NTLM by company policy. And with NTLM it works. But as soon as NTLM is deactivad and Kerberos is required we recieve the above mentioned error.
